### ISSUE TITLE
Offer a proper and secure PIN reset instead of an undocumented hack

### ISSUE THREAD

**Issue by patlecat** [nextcloud/android#4308]
### Is your feature request related to a problem? Please describe.
There is no proper way to reset the PIN and not hint nor decumentation on how to do so. Re-installation is more a hack than a feature and only a harassment to your users.

### Describe the solution you'd like
Offer at least a reset of the PIN and by doing so also clear the password of course so it can be ensured that only the right vault user can reset it.

### Describe alternatives you've considered
Offer an optional feature to reset the PIN by the NC server admin only. This way any attempts to hack/scam the user or his phone can be ruled out. Even if someone physically accesses the unprotected phone and would request a reset or re-install it themselves.

Refer to issue #4306

**Comment by mr700** [comment]
I think an option to enter the user's password instead of the pin should be good enough solution. If someone knows your password the pin does not provide extra security (one can clear the app data and set it up again or simply login from another device/browser).
Involving the admin in the matter seems too much to me... Just my 2 cents.

**Comment by patlecat** [comment]
@mr700 
Yes I see your point. I just think that adding some extra but optional security level should be a priority for a personal online vault in the times of hackers and untrustful government services. And it should be made easy to understand and use so many people would actually use it.
